### PR TITLE
Upgrade to 2.0.0.dev7

### PR DESCRIPTION
**Pull request by Eric-Arellano**
Remaining followups:

* [ ] Fix `./pants run helloworld/main.py` failing due to import error. Test it in CI.
* [ ] Turn on dependency inference. Needs docs update.
* [ ] Enable MyPy.
* [ ] Possibly change Protobuf structure to show off the new `python_source_root` plugin field. Needs docs update.
* [ ] Use `resolve_all_constraints`. Needs docs update.
* [ ] Fix all references to readme.io. Fix the `master` branch too.
* [ ] Update the `./pants` script. Fix the `master` branch too.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,3 +23,7 @@ install:
 script:
   - ./pants --changed-since=origin/master lint
   - ./pants --changed-since=origin/master --changed-include-dependees=transitive test
+  # Smoke test that our release process will work.
+  - ./pants binary helloworld/main.py helloworld/main_py2.py
```
Comment (benjyw) [inline .travis.yml:27]: We should also actually run these binaries (once they run successfully again).

File: helloworld/util/BUILD [hunk 2]
```diff
@@ -1,32 +1,37 @@
 # Copyright 2020 Pants project contributors.
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
+# See https://pants.readme.io/docs/python-setup-py-goal.
+python_distribution(
+  name="util-dist",
```
Comment (benjyw) [inline helloworld/util/BUILD:6]: Just `dist` is probably better, to avoid the stutter.

File: pants.ci.toml [hunk 3]
```diff
@@ -6,7 +9,7 @@ pantsd = false
 # Limit the maximum number of concurrent processes. Change this
 # to a number that makes sense for your CI setup, based on
 # the number of cores/threads.
-process_execution_local_parallelism = 4
+process_execution_local_parallelism = 2
```
Comment (benjyw) [inline pants.ci.toml:12]: Why not just omit this entirely? The default is probably sound. Or at least comment that the default is equal to the number of cores reported by the system.

File: .travis.yml (hunk 1 above)
Comment (Eric-Arellano) [reply .travis.yml:27]: Yes, agreed. This is a TODO in the PR description.

File: pants.ci.toml (hunk 3 above)
Comment (Eric-Arellano) [reply pants.ci.toml:12]: Because this is likely something people will want to tune, and we want to highlight that it's an option.

**Pull request merged**